slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala

```diff
@@ -59,6 +59,65 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
     )
   }
 
+  def testInsertAllTupleWithSingleInsert = ifCap(jcap.insertMultipleRowsWithSingleStatement) {
+    class A(tag: Tag) extends Table[(Int, Int)](tag, "insert_all") {
+      def id = column[Int]("id", O.PrimaryKey)
+      def v1 = column[Int]("v1")
+      def * = (id, v1)
+    }
+    val as = TableQuery[A]
+    val records = Seq((1, 10), (2, 20), (3, 30))
+    DBIO.seq(
+      as.schema.create,
+      as.insertAll(records, StatementOption.SingleStatement).map(_ shouldBe Some(3)),
+      as.sortBy(_.id).result.map(_ shouldBe records)
+    )
+  }
+  def testInsertAllProduct = ifCap(jcap.insertMultipleRowsWithSingleStatement) {
+    case class E(id: Int, v1: Int)
+    class A(tag: Tag) extends Table[E](tag, "insert_all_product") {
+      def id = column[Int]("id", O.PrimaryKey)
+      def v1 = column[Int]("v1")
+      def * = (id, v1).mapTo[E]
+    }
+    val as = TableQuery[A]
+    val records = Seq(E(1, 10), E(2, 20), E(3, 30))
+    DBIO.seq(
+      as.schema.create,
+      as.insertAll(records, StatementOption.SingleStatement).map(_ shouldBe Some(3)),
+      as.sortBy(_.id).result.map(_ shouldBe records)
+    )
+  }
+  def testInsertAllAutoInc = ifCap (jcap.returnInsertKey, jcap.insertMultipleRowsWithSingleStatement) {
+    class A(tag: Tag) extends Table[(Int, Int)](tag, "insert_all_auto_inc") {
+      def id = column[Int]("ID", O.PrimaryKey, O.AutoInc)
+      def v1 = column[Int]("V1")
```

Comment:
https://github.com/slick/slick/blob/main/slick/src/main/scala/slick/jdbc/DB2Profile.scala#L38-L42

There seems to be a bug in the DB2 driver that when using `returning` to return a column value, the column name must be uppercase.